### PR TITLE
multi: bump lnd compile time dependency to v0.18.4-beta

### PR DESCRIPTION
**Pull request by guggero**
Updates to the (not-yet-merged) `lnd v0.18.4-beta` version.

Depends on https://github.com/lightningnetwork/lnd/pull/8960.
Depends on https://github.com/lightninglabs/lndclient/pull/192.

Full serialized dependency list can be seen here: https://github.com/lightninglabs/lightning-terminal/pull/848

#### Pull Request Checklist
- [ ] Update `release_notes.md` if your PR contains major features, breaking changes or bugfixes

### CODE REVIEW

**Comment by bhandras** [comment]
This can be reviewed now?

**Comment by guggero** [comment]
> This can be reviewed now?

No, we need an RC of `lnd 0.18.4` first, see list in https://github.com/lightninglabs/lightning-terminal/pull/848.

File: go.mod [hunk 1]
```diff
@@ -20,10 +20,10 @@ require (
 	github.com/jessevdk/go-flags v1.4.0
 	github.com/lib/pq v1.10.9
 	github.com/lightninglabs/aperture v0.3.2-beta
-	github.com/lightninglabs/lndclient v0.18.0-1
+	github.com/lightninglabs/lndclient v0.18.4-0
 	github.com/lightninglabs/loop/looprpc v1.0.0
 	github.com/lightninglabs/loop/swapserverrpc v1.0.8
-	github.com/lightningnetwork/lnd v0.18.3-beta.rc1
+	github.com/lightningnetwork/lnd v0.18.3-beta.rc3.0.20241011124628-ca3bde901eb8
```
Comment (bhandras) [inline go.mod:26]: Did you mean v0.18.4-beta?

Comment (guggero) [reply go.mod:26]: We don't have a tag for that yet. This is a commit in https://github.com/lightningnetwork/lnd/pull/9183, which should remain stable (meaning I won't rebase that branch, only add more commits from new cherry-picked commits).
So we'll need another PR to bump to the final, tagged version of `lnd 0.18.4-beta`, but that hopefully will not include any other code changes, only a `go.mod/go.sum` bump.

**Review by bhandras — APPROVED** [review]
LGTM, thank you! 🎉

**Review by starius — COMMENTED** [review]
LGTM :rocket:

**Review by starius — APPROVED** [review]

File: loopout_test.go [hunk 2]
```diff
@@ -886,11 +887,15 @@ func testFailedOffChainCancelation(t *testing.T) {
 		swap.LoopOutContract.SwapInvoice, lnd.ChainParams,
 	)
 	require.NoError(t, err)
-	require.NotNil(t, invoice.PaymentAddr)
+
+	payAddr, err := invoice.PaymentAddr.UnwrapOrErr(
+		fmt.Errorf("expected payment address"),
+	)
+	require.NoError(t, err)
```
[the remark this answers is not in the document]
Comment (guggero) [reply loopout_test.go]: Ah, I forgot `UnwrapOrFail()` existed, makes perfect sense here. Changed.

**Pull request merged**